freva-rest/src/freva_rest/auth.py

```diff
@@ -127,4 +127,26 @@ async def fetch_or_refresh_token(
             logger.error(error)
             raise HTTPException(status_code=404)
         token_data = await response.json()
-    return Token(**token_data)
+        expires_at = (
```

Comment:
hmm, I'm not against of defining so many ORs for different `expires` keywords, but if you really would like to make it scaleable for the future, go for the approach that I mentioned in first review and defining a expiration_keys dictionary. 

https://github.com/FREVA-CLINT/freva-nextgen/pull/54#discussion_r1703859375

Reply:
That's not really true. The piece of code you posted, a) loops every time through keys and then b) checks for each entry if the key is within the in hash map (aka dict) that's. Under the hood that's another loop. And hence O(n^2). Retrieving items from a hash map is O(1) checking their presence O(n). So it's always better to access from a hashmap rather than checking the presence in a hashmap. Take a look at this:


```python
 def foo_get():
    lookup = {"expires_in": 200}
    return lookup.get("exp") or lookup.get("expires") or lookup.get("expires_at") or lookup.get("expires_in")

def foo_loop():
   lookup = {"expires_in": 200}
   keys = ["exp", "expires", "expires_at", "expires_in"]
   for key in keys:
      if key in lookup:
         return lookup[key]
         
%timeit foo_get()
460 ns ± 53.5 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

%timeit foo_loop()
726 ns ± 128 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```